### PR TITLE
[MI-2524] Fix and add testcases on branch MI-2474

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
Added testcases for comment modal

### CODE REVIEW

File: server/constants/constants.go [hunk 1]
```diff
@@ -162,7 +162,7 @@ const (
 	PipelineRequestContextProjectName  = "projectName"
 	PipelineRequestContextRequestType  = "requestType"
 	PipelineRequestContextRequestName  = "requestName"
-	PipelineRequestContextProjectID    = "projectID"
+	PipelineRequestContextProjectID    = "projectId"
```
Comment (ayusht2810) [inline server/constants/constants.go:165]: why this change?

File: server/plugin/api_test.go [hunk 2]
```diff
@@ -1110,135 +1071,165 @@ func TestHandlePipelineApproveOrRejectReleaseRequest(t *testing.T) {
...
 			body: `{
 				"post_id": "mockPostID",
 				"channel_id": "mockChannelID",
 				"context": {
-				  "approvalId": 1234,
+				  "approvalId": "mockApprovalID",
+				  "requestName": "run",
 				  "organization": "mockOrganization",
-				  "projectName": "mockProjectName",
+				  "projectId": "mockProjectName",
```
Comment (ayusht2810) [inline server/plugin/api_test.go:1209]:
```suggestion
				  "projectId": "mockProjectID",
```

File: server/plugin/api_test.go [hunk 3]
```diff
@@ -1110,135 +1071,165 @@ func TestHandlePipelineApproveOrRejectReleaseRequest(t *testing.T) {
...
 				"post_id": "mockPostID",
 				"channel_id": "mockChannelID",
 				"context": {
-				  "approvalId": 1234,
-				  "wrong: "mockOrganization",
-				  "projectName": "mockProjectName",
+				  "approvalId": "mockApprovalID",
+				  "requestName": "run",
+				  "organization": "mockOrganization",
+				  "projectId": "mockProjectName",
```
Comment (ayusht2810) [inline server/plugin/api_test.go:1194]: same

File: server/plugin/client_test.go [hunk 4]
```diff
@@ -318,6 +319,43 @@ func TestDeleteSubscription(t *testing.T) {
...
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			monkey.PatchInstanceMethod(reflect.TypeOf(&client{}), "Call", func(_ *client, basePath, method, path, contentType, mattermostUserID string, inBody io.Reader, out interface{}, formValues url.Values) (responseData []byte, statusCode int, err error) {
+				return nil, testCase.statusCode, testCase.err
+			})
+
+			statusCode, err := p.Client.OpenDialogRequest(&model.OpenDialogRequest{}, testutils.MockMattermostUserID)
+
+			if testCase.err != nil {
+				assert.Error(t, err)
```
Comment (ayusht2810) [inline server/plugin/client_test.go:349]: check error if possible

File: server/constants/constants.go (hunk 1 above)
Comment (raghavaggarwal2308) [reply server/constants/constants.go:165]: To make it consistent with other constants

**Review by ayusht2810 — APPROVED** [review]

File: server/plugin/api_test.go [hunk 5]
```diff
@@ -941,82 +952,37 @@ func TestHandlePipelineApproveOrRejectRunRequest(t *testing.T) {
...
-				"channel_id": "mockChannelID",
-				"context": {
-				  "approvalId": "mockApproverID",
-				  "organization": "mockOrganization",
-				  "projectID": "mockProjectID",
-				  "requestType": "mockRequestType"
-				}
-			  }`,
+			description:                            "HandlePipelineApproveOrRejectRunRequest: approved/rejected the request successfully but failed to update post",
+			body:                                   body,
```
Comment (Kshitij-Katiyar) [inline server/plugin/api_test.go:963]: is body same for all test case if yes remove it and if its wrong for just one case have a bool valid body or not

File: server/plugin/api_test.go [hunk 6]
```diff
@@ -1110,135 +1071,165 @@ func TestHandlePipelineApproveOrRejectReleaseRequest(t *testing.T) {
...
-				"channel_id": "mockChannelID",
-				"context": {
-				  "approvalId": 1234,
-				  "organization": "mockOrganization",
-				  "projectName": "mockProjectName",
-				  "requestType": "mockRequestType"
-				}
-			  }`,
+			description:                         "HandlePipelineApproveOrRejectReleaseRequest: valid",
+			body:                                body,
```
Comment (Kshitij-Katiyar) [inline server/plugin/api_test.go:1075]: same

File: server/plugin/api_test.go [hunk 7]
```diff
@@ -1110,135 +1071,165 @@ func TestHandlePipelineApproveOrRejectReleaseRequest(t *testing.T) {
...
 			updatePipelineApprovalRequestStatus:    http.StatusBadRequest,
 			updatePipelineReleaseApprovalPostError: errors.New("failed to update post"),
 			statusCode:                             http.StatusInternalServerError,
 		},
 		{
-			description: "HandlePipelineApproveOrRejectReleaseRequest: failed to approve/reject the request and fetch approval details",
+			description:                         "HandlePipelineApproveOrRejectReleaseRequest: failed to approve/reject the request and fetch approval details",
+			body:                                body,
+			updatePipelineApprovalRequestStatus: http.StatusBadRequest,
+			getApprovalDetailsError:             errors.New("failed to get approval details"),
```
Comment (Kshitij-Katiyar) [inline server/plugin/api_test.go:1104]:
```suggestion
			getApprovalDetailsError:             errors.New("failed to get the approval details"),
```

File: server/plugin/api_test.go [hunk 8]
```diff
@@ -1110,135 +1071,165 @@ func TestHandlePipelineApproveOrRejectReleaseRequest(t *testing.T) {
...
-				  "projectName": "mockProjectName",
+				  "projectId": "mockProjectID",
 				  "requestType": "mockRequestType"
 				}
 			  }`,
-			updatePipelineApprovalRequestStatus:       http.StatusInternalServerError,
-			statusCode:                                http.StatusInternalServerError,
-			updatePipelineReleaseApprovalRequestError: errors.New("failed to update the pipeline approval request"),
+			statusCode:     http.StatusInternalServerError,
+			openModalError: errors.New("failed to open comment modal"),
```
Comment (Kshitij-Katiyar) [inline server/plugin/api_test.go:1214]: check this everywhere
```suggestion
			openModalError: errors.New("failed to open the comment modal"),
```

File: server/plugin/api_test.go [hunk 9]
```diff
@@ -1110,135 +1071,165 @@ func TestHandlePipelineApproveOrRejectReleaseRequest(t *testing.T) {
...
 				  "requestType": "mockRequestType"
 				}
 			  }`,
-			isPayloadInvalid: true,
-			statusCode:       http.StatusInternalServerError,
+			statusCode: http.StatusOK,
 		},
 		{
-			description: "HandlePipelineApproveOrRejectReleaseRequest: failed to approve/reject the request due to server error",
+			description: "HandlePipelineCommentModal: failed to open comment modal",
```
Comment (Kshitij-Katiyar) [inline server/plugin/api_test.go:1201]:
```suggestion
			description: "HandlePipelineCommentModal: failed to open the comment modal",
```

**Review by avas27JTG — APPROVED** [review]

**Review by Kshitij-Katiyar — APPROVED** [review]
